ISPN-13020 Datasource  CLI test (#9331)
infinispan/infinispan: documentation/src/main/asciidoc/topics/proc_server_testing_datasources.adoc

[id='testing_datasources']
= Testing Data Sources
Data Sources can be tested via the CLI to verify that the connection to the database is functioning correctly.

.Procedure

> . Open a terminal in `{server_home}`.

Reviewer: I'll ask @oraNod to review the docs :) 

but I guess you mean "start the CLI" since "terminal" seems like it mentions the Linux terminal :)

Author: I've amended this part

> [id='testing_datasources']

Reviewer 2: ```suggestion
[id='testing-datasources']
```

Reviewer 2: I probably need to add this to the contributor's guide but dash is better than underscore in anchors for SEO purposes.

= Testing Data Sources

> Data Sources can be tested via the CLI to verify that the connection to the database is functioning correctly.

Reviewer 2: ```suggestion
Verify that connections to data sources are functioning correctly with the CLI.
```

Reviewer 2: avoid passive voice and put focus on the user goal which is to "verify data source connection" rather than test via the CLI.

.Procedure

. Start the CLI.
+
[source,options="nowrap",subs=attributes+]
----
$ bin/cli.sh
[disconnected]>
----
+
. List all data sources:
+
[source,options="nowrap",subs=attributes+]
----
[//containers/default]> server datasource ls
----
+
. Test a data source connection.
+
[source,options="nowrap",subs=attributes+]
----
[//containers/default]> server datasource test my-datasource

> ----

Reviewer 2: Could we provide a hint for resolving connection issues? 

I can open a separate docs enhancement for this if you want to get this merged sooner. But I feel like we could expand this out with some basic troubleshooting.

Reviewer 2: although maybe those troubleshooting hints could go into logging descriptions.

Author: I've added more details to the failure message

Reviewer 2: nice!
